lib/rsolr/document.rb

```diff
@@ -38,7 +38,11 @@ def field_by_name(name)
     def add_field(name, values, options = {})
       wrap(values).each do |v|
         next if v.nil?
-        @fields << RSolr::Field.instance(options.merge({:name=>name}), v)
+        if name.to_s == '_childDocuments_'
+          @fields << RSolr::Field.instance(options.merge({name: name, type: DocumentField }), v)
+        else
+          @fields << RSolr::Field.instance(options.merge({:name=>name}), v)
```

Comment:
These two lines are the same except `type:` in the hash. Can we put the hash construction in the conditional, and push the common part outside?